e2e-test/app_helper.py

```diff
@@ -48,21 +48,26 @@ def poll_for_app_url(workspaceId, app_type, proxy_url_name, azure_token, leo_url
                 # Don't run in an infinite loop if you forgot to start the app/it was never created
                 if app_type not in [item['appType'] for item in response]:
                     logging.warning(f"{app_type} not found in apps, has it been started?")
-                    return ""
-                for entries in response:
-                    if entries['appType'] == app_type:
-                        if entries['status'] == "PROVISIONING":
-                            logging.info(f"{app_type} is still provisioning. Sleeping for 30 seconds")
-                            time.sleep(30)
-                        elif entries['status'] == 'ERROR':
-                            logging.error(f"{app_type} is in ERROR state. Error details: {entries['errors']}")
-                            return ""
-                        elif entries['proxyUrls'][proxy_url_name] is None:
-                            logging.error(f"{app_type} proxyUrls not found: {entries}")
-                            return ""
-                        else:
-                            logging.info(f"{app_type} is in READY state")
-                            return entries['proxyUrls'][proxy_url_name]
+                    # Give the app time to appear in the list
+                    if polling_attempts_remaining < 10:
```

Comment:
did you mean `< 1` here?

Reply:
No, I don't want to go through ALL the polling attempts before quitting.  Really, I probably only need to let one or two go by (so it would be `< 28` or something) before the app shows up in the list, I was just being safe as well as picking a nice round number.

Reply:
ah gotcha, that might be worth a code comment but I am ok with the implementation, thank you for the explanation